### ISSUE TITLE
Support for setting a JobIdentifier for a job

### ISSUE THREAD

**Issue by StrauneanuDimitri** [vapor/queues#66]
Currently, there is no way to inject a `JobIdentifier` for a certain job using `queue.dispatch(...)` so that we can use that `JobIdentifier` to clear a job from the queue, by calling `queue.clear(id)`.

**Comment by mcdappdev** [comment]
This is an interesting idea. Perhaps the dispatch method should just return the JobIdentifier that it creates to keep it consistent. @tanner0101 what do you think?

**Comment by StrauneanuDimitri** [comment]
@mcdappdev - How about passing the `JobIdentifier` without breaking changes? This is what I currently use (I have made a new Queue extension):
```public func dispatch<J>(
        _ job: J.Type,
        _ id: JobIdentifier = JobIdentifier(),
        _ payload: J.Payload,
        maxRetryCount: Int = 0,
        delayUntil: Date? = nil
    ) -> EventLoopFuture<Void>
        where J: Job
    {
        let bytes: [UInt8]
        do {
            bytes = try J.serializePayload(payload)
        } catch {
            return self.eventLoop.makeFailedFuture(error)
        }
        let storage = JobData(
            payload: bytes,
            maxRetryCount: maxRetryCount,
            jobName: J.name,
            delayUntil: delayUntil,
            queuedAt: Date()
        )
        return self.set(id, to: storage).flatMap {
            self.push(id)
        }.map { _ in
            self.logger.info("Dispatched queue job", metadata: [
                "job_id": .string("\(id)"),
                "job_name": .string(job.name),
                "queue": .string(self.queueName.string)
            ])
        }
    }```

**Comment by mcdappdev** [comment]
@StrauneanuDimitri I like that syntax, I just pushed to the PR with it :)

**Comment by mcdappdev** [comment]
Merged in #64
